metric/system/process/process_linux_common.go

```diff
@@ -121,64 +120,81 @@ func FillPidMetrics(hostfs resolve.Resolver, pid int, state ProcState, filter fu
 		return state, fmt.Errorf("error getting metadata for pid %d: %w", pid, err)
 	}
 
-	//username
+	// username
 	state.Username, err = getUser(hostfs, pid)
 	if err != nil {
 		return state, fmt.Errorf("error creating username for pid %d: %w", pid, err)
 	}
 	return state, nil
 }
 
-// GetInfoForPid fetches the basic hostinfo from /proc/[PID]/stat
-func GetInfoForPid(hostfs resolve.Resolver, pid int) (ProcState, error) {
-	path := hostfs.Join("proc", strconv.Itoa(pid), "stat")
-	data, err := ioutil.ReadFile(path)
+// GetInfoForPid fetches and parses the process information of the process
+// identified by pid from /proc/[PID]/stat
+func GetInfoForPid(hostFS resolve.Resolver, pid int) (ProcState, error) {
+	path := hostFS.Join("proc", strconv.Itoa(pid), "stat")
+	data, err := os.ReadFile(path)
 	// Transform the error into a more sensible error in cases where the directory doesn't exist, i.e the process is gone
 	if err != nil {
 		if os.IsNotExist(err) {
 			return ProcState{}, syscall.ESRCH
 		}
 		return ProcState{}, fmt.Errorf("error reading procdir %s: %w", path, err)
+	}
 
+	state, err := parseProcStat(data)
+	state.Pid = opt.IntWith(pid)
+	if err != nil {
```

Comment:
Why do we check the error after we assign to the `state`?

Reply:
it needs to be done regardless of the error. `parseProcStat` does not do that, but it returns the most data it can get, so the best moment to set it is after `parseProcStat` returns, that way I can add the PID to the `state` it returned.